addons/stock_account/models/stock_picking.py

```diff
@@ -12,9 +12,10 @@ class StockPicking(models.Model):
 
     @api.constrains("scheduled_date", "date_done")
     def _check_backdate_allowed(self):
-        for picking in self:
-            if picking._is_date_in_lock_period():
-                raise ValidationError(self.env._("You cannot modify the scheduled date of this operation because it falls within a locked fiscal period."))
+        if not modules.module.current_test:
+            for picking in self:
+                if picking._is_date_in_lock_period():
```

Comment:
```suggestion
        if self.env['ir.config.parameter'].sudo()._get_params('stock_account.skip_lock_date_check"):
            return
        for picking in self:
            if picking._is_date_in_lock_period():
```
not tested but something similar

Reply:
@amoyaux I tested it and it's working well too :+1: